src/phoenix/server/api/helpers/dataset_helpers.py

```diff
@@ -133,3 +137,42 @@ def _get_generic_io_value(
     ):
         return io_value
     return {}
+
+
+def _get_message(message: Mapping[str, Any]) -> Dict[str, Any]:
+    content = get_attribute_value(message, MESSAGE_CONTENT)
+    name = get_attribute_value(message, MESSAGE_NAME)
+    function_call_name = get_attribute_value(message, MESSAGE_FUNCTION_CALL_NAME)
+    function_call_arguments = get_attribute_value(message, MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON)
+    function_call = (
+        {"name": function_call_name, "arguments": function_call_arguments}
+        if function_call_name is not None or function_call_arguments is not None
+        else None
+    )
+    tool_calls = [
+        {
+            "function": {
+                "name": get_attribute_value(tool_call, TOOL_CALL_FUNCTION_NAME),
+                "arguments": get_attribute_value(tool_call, TOOL_CALL_FUNCTION_ARGUMENTS_JSON),
+            }
+        }
+        for tool_call in get_attribute_value(message, MESSAGE_TOOL_CALLS) or ()
+    ]
+    return {
+        "role": get_attribute_value(message, MESSAGE_ROLE),
+        **({"content": content} if content is not None else {}),
+        **({"name": name} if name is not None else {}),
+        **({"function_call": function_call} if function_call is not None else {}),
+        **({"tool_calls": tool_calls} if tool_calls else {}),
+    }
+
+
+MESSAGE_CONTENT = MessageAttributes.MESSAGE_CONTENT
+MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON = MessageAttributes.MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON
+MESSAGE_FUNCTION_CALL_NAME = MessageAttributes.MESSAGE_FUNCTION_CALL_NAME
+MESSAGE_NAME = MessageAttributes.MESSAGE_NAME
+MESSAGE_ROLE = MessageAttributes.MESSAGE_ROLE
+MESSAGE_TOOL_CALLS = MessageAttributes.MESSAGE_TOOL_CALLS
+
+TOOL_CALL_FUNCTION_NAME = ToolCallAttributes.TOOL_CALL_FUNCTION_NAME
+TOOL_CALL_FUNCTION_ARGUMENTS_JSON = ToolCallAttributes.TOOL_CALL_FUNCTION_ARGUMENTS_JSON
```

Comment:
Is there overlap between this and the OpenAI-ft code?

Reply:
that one just plucks out the messages. 

`tool_calls` is actually not yet supported by fine-tuning, but i didn't reject it